railties/lib/rails/application/configuration.rb

```diff
@@ -449,6 +450,18 @@ def database_configuration
         raise e, "Cannot load database configuration:\n#{e.message}", e.backtrace
       end
 
+      def autoload_lib(ignore:)
```

Comment:
@fxn  Maybe by calling this method, we can also remove `lib` from Ruby's load path https://github.com/rails/rails/blob/5c5b78dfa7d4ab0db884d10d3a4f02f5883c9b68/railties/lib/rails/application.rb#L387**

only if this [option](https://guides.rubyonrails.org/configuring.html#config-add-autoload-paths-to-load-path) is set to false

Reply:
@chaadow good call. I think this one is tricky:

* People may expect `lib` to be in `$LOAD_PATH` because it is generally there in Ruby projects.
* Rails issues a `require` to load generators on demand. `lib/generators` is idiomatic and you'd expect that to work as it did until now.

Let me think about it. Even if we special-case it, it'd need some docs and tests, so the comment was useful in either case, thanks!

Reply:
In the end, I opted for documenting and adding test coverage for it (https://github.com/rails/rails/pull/48596). Thanks!